packages/lab/src/dialog/Dialog.tsx

```diff
@@ -89,49 +114,39 @@ export const Dialog = forwardRef<HTMLDivElement, DialogProps>(function Dialog(
     [dialogId, status]
   );
 
+  if (!open && !showComponent) return null;
```

Comment:
Was the impact of this to desktop support checked?

Reply:
This is how it's done in overlay, was waiting for the feedback on overlay.
Looked into conditionally rendering just the Scrim and Children, so the FloatingComponent was always inserted into the DOM, but then not sure which element {...getFloatingProps} would have to be attached to.
If the Scrim is a child of the FloatingElement then using the inbuilt 'useDimiss' hook won't work

Reply:
this pr is to get to RC enabling testing in desktop, any necessary changes will be make in next sprint before promoting to core